### PR TITLE
feat: Support aux loss normalization in RL SFT

### PR DESCRIPTION
**Pull request by pthombre**
# What does this PR do ?
Remove the MoE aux loss assertion that blocked aux_loss usage with calculate_per_token_loss=True. Add moe_grad_scale_func to properly normalize MOE auxiliary loss gradients: sets scale to 1/global_valid_toks before forward-backward and clears it after, so that after DDP SUM the aux loss gradient is correctly averaged.

Also adds sft_nanov3.yaml config for nano-v3 SFT training with MoE seq_aux_loss enabled.

# Issues
List issues that this PR closes ([syntax](https://docs.github.com/en/issues/tracking-your-work-with-issues/using-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)):

# Usage
* **You can potentially add a usage example below**

```python
# Add a code snippet demonstrating how to use this
```

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](/NVIDIA-NeMo/RL/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you run the unit tests and functional tests locally? Visit our [Testing Guide](/NVIDIA-NeMo/RL/blob/main/docs/testing.md) for how to run tests
- [ ] Did you add or update any necessary documentation? Visit our [Document Development Guide](/NVIDIA-NeMo/RL/blob/main/docs/documentation.md) for how to write, build and test the docs.

# Additional Information
* ...

### CODE REVIEW

File: nemo_rl/models/policy/workers/megatron_policy_worker.py [hunk 1]
```diff
@@ -690,6 +701,21 @@ def train(
                 metrics["moe_metrics"] = moe_metrics
         return metrics
 
+    def _get_model_config(self):
```
Comment (yfw) [inline nemo_rl/models/policy/workers/megatron_policy_worker.py]: [`megatron_policy_worker.py:704`](https://github.com/NVIDIA-NeMo/RL/blob/8add7735b2a64ad3939f5e064811c1e5d8211e3a/nemo_rl/models/policy/workers/megatron_policy_worker.py#L704)

`_get_model_config` reimplements the already-imported [`get_model_config`](https://github.com/NVIDIA-NeMo/RL/blob/8add7735b2a64ad3939f5e064811c1e5d8211e3a/nemo_rl/models/policy/workers/megatron_policy_worker.py#L46) (used at [L302](https://github.com/NVIDIA-NeMo/RL/blob/8add7735b2a64ad3939f5e064811c1e5d8211e3a/nemo_rl/models/policy/workers/megatron_policy_worker.py#L302) / [L383](https://github.com/NVIDIA-NeMo/RL/blob/8add7735b2a64ad3939f5e064811c1e5d8211e3a/nemo_rl/models/policy/workers/megatron_policy_worker.py#L383) / [L615](https://github.com/NVIDIA-NeMo/RL/blob/8add7735b2a64ad3939f5e064811c1e5d8211e3a/nemo_rl/models/policy/workers/megatron_policy_worker.py#L615)). The hand-rolled `hasattr` walk returns `None` on no-match, so `_set_moe_grad_scale_func` silently no-ops and the aux gradient falls back to un-normalized **with no error**. It resolves correctly for the single-DDP path today, but it's a silent footgun if VPP / a model-chunk list is ever enabled for this worker. Consider reusing `get_model_config(self.model)` (which raises on a list rather than silently skipping) and dropping `_get_model_config`.

Minor: add type hints (`func: Optional[Callable[[], torch.Tensor]]`, `-> None`), and consider wrapping the set/clear around `megatron_forward_backward` in `try/finally` so a raised forward-backward doesn't leave a stale scale installed for a later forward.

File: nemo_rl/models/megatron/setup.py [hunk 2]
```diff
@@ -884,14 +884,9 @@ def _validate_training_config(config: PolicyConfig, model_cfg: Any) -> None:
...
 
-    # MoE aux loss validation
-    assert (
-        "aux_loss" not in model_cfg.moe_router_load_balancing_type
-        or model_cfg.moe_aux_loss_coeff == 0
-    ), (
-        "MoE aux loss is currently not supported due to a known bug in Megatron-LM. "
-        "See https://github.com/NVIDIA/Megatron-LM/issues/1984 for more details."
-    )
+    # MoE aux loss validation - disabled to support aux loss normalization in RL SFT.
```
Comment (yfw) [inline nemo_rl/models/megatron/setup.py:887]: [`setup.py:887`](https://github.com/NVIDIA-NeMo/RL/blob/c11b636264229dbf82f07ee497f0c247f25c3282/nemo_rl/models/megatron/setup.py#L887)

**This breaks an existing test (CI).** `tests/unit/models/megatron/test_megatron_setup.py::test_moe_aux_loss_not_supported` still does `pytest.raises(AssertionError)` expecting `"MoE aux loss is currently not supported"`, which no longer fires — the `@pytest.mark.mcore` lane will go red. Could you delete or flip that test (e.g. assert aux loss is now allowed)? [test source](https://github.com/NVIDIA-NeMo/RL/blob/c11b636264229dbf82f07ee497f0c247f25c3282/tests/unit/models/megatron/test_megatron_setup.py#L1258)

File: examples/configs/sft_nanov3.yaml [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+      lr_warmup_init: 3e-7 #0.0
+
+    distributed_data_parallel_config:
+      grad_reduce_in_fp32: true #false
+      overlap_grad_reduce: true
+      overlap_param_gather: true
+      data_parallel_sharding_strategy: "none"  #"optim_grads_params"
+      use_custom_fsdp: false
+
+data:
```
Comment (yfw) [inline examples/configs/sft_nanov3.yaml]: [`sft_nanov3.yaml:157`](https://github.com/NVIDIA-NeMo/RL/blob/c11b636264229dbf82f07ee497f0c247f25c3282/examples/configs/sft_nanov3.yaml#L157)

This config is auto-discovered by `tests/unit/test_config_validation.py` (it globs `examples/configs/**/*.yaml`, [L95](https://github.com/NVIDIA-NeMo/RL/blob/c11b636264229dbf82f07ee497f0c247f25c3282/tests/unit/test_config_validation.py#L95), and is `run_first`), and it fails `SFTMasterConfig` validation → unit CI will be red. It uses the older flat data schema; the current schema needs nested `data.train`/`data.validation`. Specifically:

- `data.train` required (config has flat `train_data_path`/`val_data_path` at [L165-166](https://github.com/NVIDIA-NeMo/RL/blob/c11b636264229dbf82f07ee497f0c247f25c3282/examples/configs/sft_nanov3.yaml#L165))
- `sft.val_at_end` required (missing)
- `sft.only_unmask_final` required (missing)
- `checkpointing.keep_top_k` must be an int (config sets `null` at [L20](https://github.com/NVIDIA-NeMo/RL/blob/c11b636264229dbf82f07ee497f0c247f25c3282/examples/configs/sft_nanov3.yaml#L20))

Comparing against [`examples/configs/sft.yaml`](https://github.com/NVIDIA-NeMo/RL/blob/c11b636264229dbf82f07ee497f0c247f25c3282/examples/configs/sft.yaml#L1) is the quickest way to align these.

File: examples/configs/sft_nanov3.yaml [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+  enabled: true
+  checkpoint_dir: "results/sft_nanov3"
+  metric_name: null # one of "val:" or "train:" followed by the metric name
+  higher_is_better: false
+  keep_top_k: null
+  save_period: 200
+  checkpoint_must_save_by: null
+
+policy:
+  model_name: /mnt/nanov3_pretrained
```
Comment (yfw) [inline examples/configs/sft_nanov3.yaml]: [`sft_nanov3.yaml:25`](https://github.com/NVIDIA-NeMo/RL/blob/c11b636264229dbf82f07ee497f0c247f25c3282/examples/configs/sft_nanov3.yaml#L25)

Even once the schema validates, this reads as an environment-specific experiment config rather than a portable exemplar: hardcoded absolute paths (`/mnt/nanov3_pretrained`, `/opt/nemo-rl/super_v3_sft/chat_template.jinja`, `/mnt/cache/nanov3_data/...` with `val_data_path == train_data_path` at [L165-166](https://github.com/NVIDIA-NeMo/RL/blob/c11b636264229dbf82f07ee497f0c247f25c3282/examples/configs/sft_nanov3.yaml#L165)), a personal wandb project (`ashors-nemotron-sft`, [L181](https://github.com/NVIDIA-NeMo/RL/blob/c11b636264229dbf82f07ee497f0c247f25c3282/examples/configs/sft_nanov3.yaml#L181)), and `num_nodes: 64` ([L194](https://github.com/NVIDIA-NeMo/RL/blob/c11b636264229dbf82f07ee497f0c247f25c3282/examples/configs/sft_nanov3.yaml#L194)). Per the [config conventions](https://github.com/NVIDIA-NeMo/RL/blob/main/.agents/contributor-skills/config-conventions/SKILL.md), recipe YAMLs belong under `examples/configs/recipes/**` with a `defaults:` key pointing at an exemplar (`uv run ./tools/config_cli.py minimize <recipe.yaml>` assigns it), or consider dropping it from this PR.

File: nemo_rl/models/policy/workers/megatron_policy_worker.py [hunk 5]
```diff
@@ -640,6 +640,13 @@ def train(
                         self.optimizer.zero_grad()
                         self._copy_main_params_to_param_buffer()
 
+                    # Set moe_grad_scale_func for MOE aux loss scaling.
+                    # With calculate_per_token_loss=True, the router pre-multiplies
+                    # aux_loss by num_local_tokens. Setting loss_scale = 1/global_valid_toks
+                    # makes the final MOE gradient correctly normalized:
+                    #   (1/G) * N_local * aux_grad -> after DDP SUM -> avg(aux_grad)
```
Comment (yfw) [inline nemo_rl/models/policy/workers/megatron_policy_worker.py]: [`megatron_policy_worker.py:644`](https://github.com/NVIDIA-NeMo/RL/blob/c11b636264229dbf82f07ee497f0c247f25c3282/nemo_rl/models/policy/workers/megatron_policy_worker.py#L644)

This comment is misleading. The router pre-multiplies by `aux_loss * num_local_tokens * tp_cp_group.size()` ([router.py#L510-512](https://github.com/NVIDIA/Megatron-LM/blob/002255075c3728fded9a2e435677840b08560d55/megatron/core/transformer/moe/router.py#L510)), and `MoEAuxLossAutoScaler` scales the *gradient*, not the loss. The omitted `tp_cp_group.size()` factor is exactly what makes `1/global_valid_toks` match the main SFT loss normalization. Suggested rewording:

```suggestion
                    # Set moe_grad_scale_func for MoE aux-loss gradient scaling.
                    # With calculate_per_token_loss=True, the router pre-multiplies
                    # the aux loss by (num_local_tokens * tp_cp_group.size()), and
                    # MoEAuxLossAutoScaler applies loss_scale to the gradient. Setting
                    # loss_scale = 1/global_valid_toks (G = global valid token count)
                    # normalizes the aux gradient consistently with the main per-token
                    # SFT loss:
                    #   (1/G) * N_local * tp_cp_size * aux_grad -> DDP SUM -> aux_grad / G
```

File: nemo_rl/models/megatron/setup.py (hunk 2 above)
Comment (pthombre) [reply nemo_rl/models/megatron/setup.py:887]: Modified the test

**Review by yfw — DISMISSED** [review]

**Review by yfw — APPROVED** [review]
